### PR TITLE
Add option to hide event

### LINKED ISSUE #2206: Favor hiding events to deleting them.
## What kind of an issue is this?

- Feature request

## What is the expected behavior?

Deleting events may cause a lot of unnecessary consequences, like deleting payments and generally losing history of what has been done.

It should therefore be possible to hide an event with a boolean setting on the event.
Hiding it should be used in most cases, but there should be a dashboard override for deleting in case it is strictly necessary.

## What is the current behavior?

The only way to make an event not visible anymore is to delete it from the database.

### PR DESCRIPTION
**Pull request by henrikhorluck**
- Hides on frontpage
- Hides on /events
- Hides on detailView

#### TODO: 

- [x] Add tests
- [x] Add option to hide event in Dashboard
## What kind of a pull request is this?

Fixes #2206 
- QA / Code Review

## Code Checklist

- [x] The code follows dotkom code style 
- [x] The code passes the defined tests
- [x] I have added tests for the code I added
- [ ] I have provided documentation for the code I added
- [x] The code is ready to be merged

## (Possible) Breaking Changes

- [x] The changes are backwards compatible

- [ ] I have updated the build configuration
- [ ] These changes requires changes to configuration in production 
    - [ ] I have applied the required changes in production
    - [ ] I cannot apply the required changes in production before this is deployed.

### CODE REVIEW

File: apps/events/models.py [hunk 1]
```diff
@@ -169,7 +173,7 @@ def can_display(self, user):
             return True
         if not user:
             return False
-        return restriction.has_access(user)
+        return restriction.has_access(user) if self.visible else False
```
Comment (oleast) [inline apps/events/models.py]: Would this not still return `True` if the event has no GroupRestriction?

Comment (henrikhorluck) [reply apps/events/models.py]: Yes, because of `if not restriction: return True` earlier in the function

Comment (oleast) [reply apps/events/models.py]: If I'm not completely wrong `visible` should have higher precedence than a GroupRestriction?

In this case, all events will be visible if they don't have a GroupRestriction, which is almost all events.

Comment (henrikhorluck) [reply apps/events/models.py]: Yes, you are correct, fixed it now.

File: apps/events/models.py [hunk 2]
```diff
@@ -114,6 +114,10 @@ class Event(models.Model):
     """Event type. Used mainly for filtering"""
     organizer = models.ForeignKey(Group, verbose_name=_('arrangør'), blank=True, null=True, on_delete=SET_NULL)
     """Committee responsible for organizing the event"""
+    visible = models.BooleanField(
+        _('Vis arrangementet utenfor Dashboard/admin-panelet'),
+        default=True,
+        help_text='Denne brukes primært for å skjule eksisterende eventer.')
```
Comment (oleast) [inline apps/events/models.py]: I think wording it as 'arrangementer' would be more correct than 'eventer'

Comment (henrikhorluck) [reply apps/events/models.py]: I agree, fixed

**Pull request merged**